Sketcher/OffsetInSketcher.FCMacro

```diff
@@ -0,0 +1,431 @@
+
+	def coincident(Element1,Point1,Element2,Point2):
+		ActiveSketch.addConstraint(Sketcher.Constraint('Coincident',Element1.GeoId,Point1,Element2.GeoId,Point2))
+
+	def equal(Element1,Element2):
+		ActiveSketch.addConstraint(Sketcher.Constraint('Equal',Element1.GeoId,Element2.GeoId))
+
+	def perpendicular(Element1,Element2):
+		ActiveSketch.addConstraint(Sketcher.Constraint('Perpendicular',Element1.GeoId,Element2.GeoId))
+
+	def pointOnObject(Element1,Point,Element2):
+		ActiveSketch.addConstraint(Sketcher.Constraint('PointOnObject',Element1.GeoId,Point,Element2.GeoId))
+
+	def symmetric(Element,MidVertexNum):
+		ActiveSketch.addConstraint(Sketcher.Constraint('Symmetric',Element.GeoId,1,Element.GeoId,2,MidVertexNum,1))
+
+	def tangent(Element1,Point1,Element2,Point2):
+		ActiveSketch.addConstraint(Sketcher.Constraint('Tangent',Element1.GeoId,Point1,Element2.GeoId,Point2))
+
+class Tools:
+#
+# diverse geometric tools 
+#
+	def createLine(Point1,Point2):
+		return Edge(Part.LineSegment(Point1,Point2))
+
+	def concentricArcOfCircle(Element,Distance,Reverse=False):
+		Arc = Element.Geo.clone()
+		Circle = Arc.Circle
+		ConcentricEdge = Edge(Part.ArcOfCircle(Part.Circle(Circle,Distance),Arc.FirstParameter,Arc.LastParameter))
+		if Reverse: ConcentricEdge.reverse() # change sense of rotation
+		return ConcentricEdge
+
+	def getMidPoint(Element):
+		XPoint = (Element.EndPoint.x+Element.StartPoint.x)/2
+		YPoint = (Element.EndPoint.y+Element.StartPoint.y)/2
+		return App.Vector(XPoint,YPoint,0)
+
+class OffsetInSketcher():
+#
+# class containing the addOffset command
+#
+	def addOffset(SelGeoId,OffsetLine):
+	#
+	# this is the OffsetInSketcher command
+	#
+		def getSelectedEdges(SelGeoID):
+		#
+		# returns a list of all selected lines and arcs
+		#
+			AllGeo = Sketch.getGeometry() # full geometry list
+			Selected = []
+			for Id in SelGeoID:
+				Goemetry = AllGeo[Id]
+				Selected.append(Edge(Goemetry,Id))
+			return Selected
+
+		def areEqual(V1,V2):
+		#
+		# check the  identity of two vectors
+		#
+			if (V1-V2).Length < 0.001: return True
+
+		def interlinkedEdges(UnsortedEdges):
+		#
+		# return an interlinked list of lines and arcs
+		#
+			def getBackwardElem(Element,UnsortedEdges):
+			#
+			# get next edge in backward direction
+			#
+				if len(UnsortedEdges) == 0: return
+				else:
+					StartPt = Element.StartPoint
+					Num = 0
+					while Num <= len(UnsortedEdges)-1: # search backward
+						El=UnsortedEdges[Num]
+						if (areEqual(El.StartPoint,StartPt)) or (areEqual(El.EndPoint ,StartPt)): # element found
+							if areEqual(El.StartPoint,StartPt): # swap element
+								El.reverse() 
+							SortedEdges.append(El)
+							UnsortedEdges.pop(Num)
+							getBackwardElem(El,UnsortedEdges)
+							break
+						Num = Num+1
+					return # nothing found
+
+			def getForwardElem(Element,UnsortedEdges):
+			#
+			# get next edge in forward direction
+			#
+				if len(UnsortedEdges) == 0: return
+				else:
```

Comment:
You can remove the `else:` line and reduce the level of indentation in the following block because the `if` clause returns.